README.md

```diff
@@ -3,8 +3,8 @@ An opinioned coding guideline for React + Redux project @WORD UP
 
 ## Naming
 #### File name & Folder name
-  * Lowercase. Separated by single dash
-    * `module/components/user-feedback.js`
+  * Camel-Case, equal to component name
```

Comment:
should be better to use component name as folder name

Reply:
How about plain modules not components like `utils/Modules/Cards/fetcher.js`?

Reply:
I think use should prevent nested structure for the folder like what we do on data structure.
So my suggestion is
```
|- utils
   |- APIs
      |- CardsFetcher.js
```

Reply:
The `nested data structure` is a differenet case. Since we need to update the data frequently, it would be error-prone to have many levels. And it also relates to the update mechanism of React and Redux.

About the file structure, it is common that we'll have more and more service/utility modules when the project becomes bigger and bigger. The folder structure help to make those files organized. So the number of module files is potentially unlimited, it's hard to say that we just avoid nested structure. 

BTW. For the example: `utils/Modules/Cards/fetcher.js` It's not an `api` service, actually it handles quite complicated tasks to fetch cards from local storage or server.

But, actually, what's the benefit of `Camel-Case, equal to component name`? For me, the benefit of using `dash-separate` is consistency. It's an easy rule that suitable for all files.

Reply:
BTW (this is not directly to this naming topic, just correct the `utils/APIs/CardsFetcher.js`)  It's not an `api` service, actually it handles quite complicated tasks to fetch cards from local storage or server.

Reply:
I don't think to organized in nested structure will be easier to manage.
Maybe for single person's project, but with many people, it will be hard to keep the same logic about how to organize the files.
And I prefer to search in a large file rather than searching in a nested folder with many small files.

With `lowercase-separated-by-dash`, we will need to import component/class like this, I don't see any consistency
`import UserSetting from 'Components/user-setting';`
And if we want to rename the component, we have to rename to two different type.

I haven't seen any React project name the component folder with `lowercase-separated-by-dash`.
Check below repo for reference.
https://github.com/react-boilerplate/react-boilerplate/tree/master/app/components

Actually I don't mind to naming util folder in any format, but for class(component or fetcher), I believe to name the folder by class name will be better.

Reply:
so consensus here is
HeaderComponent
or
header-component

Reply:
@oahehc got it. Thanks! So about the naming convention of component files, seems `react-boilerplate` uses PascalCase for component and lowercase for [util files](https://github.com/react-boilerplate/react-boilerplate/tree/master/app/utils). Also, the [Airbnb styleguid](https://github.com/react-boilerplate/react-boilerplate/tree/master/app/utils) uses PascalCase for component files as well. As you said, let's follow the community convention.

Reply:
But there're some points that I don't agree with now (just discussion)

> we will need to import component/class like this, I don't see any consistency
import UserSetting from 'Components/user-setting';

`'Components/user-setting'` is file name and `UserSetting` is function/class name. They naturally don't need to be the same. 
Like `import { isEmpty } from 'lodash'`, `import { fetchCards } from '../cards-apis.js`. Or we can make a simple rule that always using PascalCase  as [here](https://github.com/Semantic-Org/Semantic-UI-React/tree/master/src/lib) and [here](https://github.com/lodash/lodash)

Reply:
```
should correct Camel-Case to be CamelCase
```
```
How about we just use PascalCase
```
PascalCase for component and camelCase for helper function right?

Reply:
oh, yes.